search-app/src/cmr/search/api/concepts_search.clj

```diff
@@ -0,0 +1,223 @@
+                        (rfh/printable-result-format result-format) (pr-str params)))
+        search-params (lp/process-legacy-psa params)
+        results (query-svc/find-concepts-by-parameters ctx concept-type search-params)]
+    (core-api/search-response ctx results)))
+
+(defn- find-concepts
+  "Invokes query service to find results and returns the response.
+
+  This function supports several cases for obtaining concept data:
+  * By JSON query
+  * By parameter string and URL parameters
+  * Collections from Granules - due to the fact that ES doesn't suport joins
+    in the way that we need, we have to make two queries here to support CMR
+    Harvesting. This can later be generalized easily, should the need arise."
+  [ctx path-w-extension params headers body]
+  (let [content-type-header (get headers (string/lower-case common-routes/CONTENT_TYPE_HEADER))]
+    (cond
+      (= mt/json content-type-header)
+      (find-concepts-by-json-query ctx path-w-extension params headers body)
+
+      (or (nil? content-type-header) (= mt/form-url-encoded content-type-header))
+      (find-concepts-by-parameters ctx path-w-extension params headers body)
+
+      :else
+      {:status 415
+       :headers {common-routes/CORS_ORIGIN_HEADER "*"}
+       :body (str "Unsupported content type ["
+                  (get headers (string/lower-case common-routes/CONTENT_TYPE_HEADER)) "]")})))
+
+(defn- get-granules-timeline
+  "Retrieves a timeline of granules within each collection found."
+  [ctx path-w-extension params headers query-string]
+  (let [params (process-params :granule params path-w-extension headers mt/json)
+        _ (info (format "Getting granule timeline from client %s with params %s."
+                        (:client-id ctx) (pr-str params)))
+        search-params (lp/process-legacy-psa params)
+        results (query-svc/get-granule-timeline ctx search-params)]
+    {:status 200
+     :headers {common-routes/CORS_ORIGIN_HEADER "*"}
+     :body results}))
+
+(defn- find-concepts-by-aql
+  "Invokes query service to parse the AQL query, find results and returns the response"
+  [ctx path-w-extension params headers aql]
+  (let [params (process-params nil params path-w-extension headers mt/xml)
+        _ (info (format "Searching for concepts from client %s in format %s with AQL: %s and query parameters %s."
+                        (:client-id ctx) (rfh/printable-result-format (:result-format params)) aql params))
+        results (query-svc/find-concepts-by-aql ctx params aql)]
+    (core-api/search-response ctx results)))
+
+(defn- find-tiles
+  "Retrieves all the tiles which intersect the input geometry"
+  [ctx params]
+  (let [results (query-svc/find-tiles-by-geometry ctx params)]
+    {:status 200
+     :headers {common-routes/CONTENT_TYPE_HEADER (mt/with-utf-8 mt/json)}
+     :body results}))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;; Route Definitions
```

Comment:
This comment should be moved to where the routes are defined or deleted

Reply:
Whoops -- bad merge/rebase. Thanks.

Fixed.